### ISSUE TITLE
GitHub App login fails with "Access denied"

### ISSUE THREAD

**Issue by timebertt** [octobox/octobox#2651]
Are you experiencing this issue in octobox.io or your own instance?
 - [ ] octobox.io
 - [x] My instance

🐞 **Problem**

I've set up an own Octobox instance, restricted access to a given organization and team and configured a GitHub App according to https://github.com/octobox/octobox/blob/master/docs/INSTALLATION.md#running-octobox-as-a-github-app.

The restricted access works fine, I'm able to login (as I'm part of the org/team) and everything looks good (OAuth flow works).
When I try to login to the GitHub App from Octobox, the authorization flow itself also works (the App is shown as an authorized App in my GitHub settings), but I get a red banner saying `Access denied.` (it seems the org/team check fails [here](https://github.com/octobox/octobox/blob/bedc8819a0e6cc101abe2fa37e3b498f10a4f0d3/app/controllers/sessions_controller.rb#L46)) and the settings page still shows the button to login to the GitHub App.

When I disable restricted access again, logging in to the GitHub App still works fine, though I don't get the red banner and the settings page reflect the login accordingly.

🎯 **Goal**

Enable to run Octobox as a GitHub App and simultaneously restrict access to a given org/team.

**Comment by andrew** [comment]
Sounds like a bug, I've not heard of anyone using the restrict access feature whilst running octobox as a github app before (i've not done it myself either)

**Comment by timebertt** [comment]
Thanks @andrew for your reply!
Is it then still a valid usecase to run Octobox as a GitHub App and still restrict access? Or am I heading in a wrong direction?

**Comment by andrew** [comment]
@timebertt yeah I think it's a valid use case, just not once that I've personally tried.
